### PR TITLE
Intergration

### PR DESCRIPTION
**Pull request by davidsalsa**

### CODE REVIEW

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 1]
```diff
@@ -128,33 +131,21 @@ public void ShowOrder()
     private IEnumerator SayOrder(string order)
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: I guess we can remove the coroutine here? As the popup only shows the image and no text. So the delay is unnecessary, plus the coroutine isn't stopped.

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 2]
```diff
@@ -258,18 +249,18 @@ public void TakeCorrectPositionInQueue()
     {
         LinkedListNode<Adventurer> adventurerInFront = _orderQueue.GetList().Find(_adventurer).Previous;
 
-        Vector3 adventurerDirection = (new Vector3(transform.position.x, 0, transform.position.z) - _deskPos.position).normalized;
+        Vector3 adventurerDirection = (new Vector3(transform.position.x, 0, transform.position.z) - new Vector3(_deskPos.position.x, 0, _deskPos.position.z)).normalized;
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: Can be simplified to `transform.position - _deskPos.position)`

File: behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class OrderPopup : MonoBehaviour, IUIElement
+{
+
+    GameObject orderImage;
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs]: _ before orderImage

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 4]
```diff
@@ -258,18 +249,18 @@ public void TakeCorrectPositionInQueue()
...
 
-        if((_queueDirection - adventurerDirection).magnitude > queueLineOffset)
+        if(Vector3.SignedAngle(_queueDirection, adventurerDirection, Vector3.up) > queueLineOffset)
         {
-            Vector3 perpendicular = new Vector3(_queueDirection.z, 0, -_queueDirection.x);
+            Vector3 perpendicular = new Vector3(-_queueDirection.z, 0, _queueDirection.x);
             _agent.SetDestination(transform.position + perpendicular * queueMovementSpeed);
         } 
-        else if ((_queueDirection - adventurerDirection).magnitude < -queueLineOffset)
+        else if (Vector3.SignedAngle(_queueDirection, adventurerDirection, Vector3.up) < -queueLineOffset)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: This calculation is performed twice, you should save it in a variable.

File: behind-the-tankard/Assets/Scripts/General/EncounterManager.cs [hunk 5]
```diff
@@ -20,18 +20,18 @@ void Update()
 
     private void DoEvent()
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/Scripts/General/EncounterManager.cs]: Does this script do anything?

File: behind-the-tankard/Assets/Scripts/General/EncounterManager.cs [hunk 6]
```diff
@@ -20,18 +20,18 @@ void Update()
...
     {
-        Adventurer chosenAdventurer = TavernManager.adventurers[Random.Range(0, TavernManager.adventurers.Count - 1)];
+        // Adventurer chosenAdventurer = TavernManager.adventurers[Random.Range(0, TavernManager.adventurers.Count - 1)];
```
Comment (Lunium) [inline behind-the-tankard/Assets/Scripts/General/EncounterManager.cs]: All this code is commented?

File: behind-the-tankard/Assets/Scripts/PlayerMovement.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+    void Start()
+    {
+        _controller = GetComponent<CharacterController>();
+    }
+
+    void Update()
+    {
+        ResetGravityWhenNearGround();
+
+        Move();
```
Comment (YWAMWeessies) [inline behind-the-tankard/Assets/Scripts/PlayerMovement.cs]: Does this not belongs in the fixedupdate?

File: behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs [hunk 8]
```diff
@@ -258,18 +249,18 @@ public void TakeCorrectPositionInQueue()
     {
         LinkedListNode<Adventurer> adventurerInFront = _orderQueue.GetList().Find(_adventurer).Previous;
 
-        Vector3 adventurerDirection = (new Vector3(transform.position.x, 0, transform.position.z) - _deskPos.position).normalized;
+        Vector3 adventurerDirection = (new Vector3(transform.position.x, 0, transform.position.z) - new Vector3(_deskPos.position.x, 0, _deskPos.position.z)).normalized;
 
         _agent.stoppingDistance = 0;
 
-        if((_queueDirection - adventurerDirection).magnitude > queueLineOffset)
+        if(Vector3.SignedAngle(_queueDirection, adventurerDirection, Vector3.up) > queueLineOffset)
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/AI/AdventurerAI.cs]: This if-else statement is quite big and contains no comments, it is very hard to read.
Document with one line comments what it is that the if should detect and what should be done.

File: behind-the-tankard/Assets/Scripts/PlayerMovement.cs (hunk 7 above)
Comment (Lunium) [reply behind-the-tankard/Assets/Scripts/PlayerMovement.cs]: The movement doesn't use a rigidbody I guess. So it's not physics based.

File: behind-the-tankard/Assets/Scripts/PlayerMovement.cs [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+    private void ResetGravityWhenNearGround()
+    {
+        if (IsCharacterGrounded())
+        {
+            _velocity.y = _groundedGravity;
+        }
+    }
+
+    private void ApplyGravity()
```
Comment (MattijnP) [inline behind-the-tankard/Assets/Scripts/PlayerMovement.cs]: Is our own implementation of gravity really necessary if we're using rigidbodies?

File: behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+    void Awake()
+    {
+        orderImage = transform.Find("OrderImage").gameObject;
+    }
+
+    void Update()
+    {
+        transform.LookAt(2 * transform.position - Camera.main.transform.position);
```
Comment (TJ-Smits) [inline behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs]: Magic number?

File: behind-the-tankard/Assets/Scripts/General/EncounterManager.cs (hunk 5 above)
Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/General/EncounterManager.cs]: Solid question

File: behind-the-tankard/Assets/Scripts/General/EncounterManager.cs (hunk 6 above)
Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/General/EncounterManager.cs]: Good question. Why is it commented?

File: behind-the-tankard/Assets/Scripts/PlayerMovement.cs (hunk 9 above)
Comment (TJ-Smits) [reply behind-the-tankard/Assets/Scripts/PlayerMovement.cs]: If we are using a rigidbody for the main character then it is necesssary. The implementation i helped with was without a rigidbody.

Comment (Lunium) [reply behind-the-tankard/Assets/Scripts/PlayerMovement.cs]: unnecesssary*

File: behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs (hunk 10 above)
Comment (Lunium) [reply behind-the-tankard/Assets/Scripts/DialogueSystem/OrderPopup.cs]: Not really. The code used to be `transform.LookAt(Camera.main.transform.position);`, but this flips the sprite images. So this code takes a position completely opposite of the player looking from the camera.

**Pull request merged**